src/core/lib/debug/stats.h

```diff
@@ -35,13 +35,20 @@ extern grpc_stats_data* grpc_stats_per_cpu_storage;
 #define GRPC_THREAD_STATS_DATA() \
   (&grpc_stats_per_cpu_storage[grpc_core::ExecCtx::Get()->starting_cpu()])
 
```

Comment:
Can just define ```GRPC_COLLECT_STATS ifndef NDEBUG```` right here so we can simplify the combined check at all the other points?

Reply:
hmm we'd need to include stats.h at exec_ctx.h then

Reply:
(which is a circular dependency). This can be separated into a separate file though which seems like an overkill. This check shouldn't snowball any further, but if you feel strongly about this, it can be done.

Reply:
I don't feel strongly about this.  If there is a dependency issue making this difficult, than I'm fine with it as is.